lib/astarte_vmq_plugin.ex

```diff
@@ -255,4 +265,17 @@ defmodule Astarte.VMQ.Plugin do
 
     "#{realm_trunc}-#{device_id_trunc}-#{timestamp_hex_str}-#{rnd}"
   end
+
+  defp get_connection_reconciler_process(client_id) do
+    case Registry.lookup(AstarteVMQPluginConnectionSynchronizer.Registry, client_id) do
+      [] ->
+        # TODO a single dynamic supervisor may become a bottleneck,
+        # use a PartitionSupervisor when on Elixir >= 1.14
```

Comment:
Move this `TODO` inside the SynchronizersSupervisor, since that's where the change will happen (the API will probably stay the same)